bigbluebutton-html5/imports/ui/components/presentation/cursor/cursor-wrapper-container/container.jsx

```diff
@@ -0,0 +1,69 @@
+/*
+  The purpose of this wrapper is to fetch an array of active cursor iDs only
+  and map them to the CursorContainer.
+  The reason for this is that Meteor tracks only the properties defined inside createContainer
+  and if we fetch the whole array of Cursors right away (let's say in the multi-user mode),
+  then the whole array would be re-rendered every time one of the Cursors is changed.
+
+  To work around this we map only an array of cursor iDs to CursorContainer.
+  This list will be updated in cases when new users join/leave the meeting.
+  And then each separate Cursorcontainer will keep
+  track of that particular Cursor properties, thus all of them will be updated independently.
+*/
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { createContainer } from 'meteor/react-meteor-data';
+import CursorWrapperService from './service';
+import CursorContainer from '../container';
+
+
+const CursorWrapperContainer = ({ presenterCursorId, multiUserCursorIds, ...rest }) => (
+  <g>
+    { presenterCursorId ?
+      <CursorContainer
+        key={presenterCursorId._id}
+        presenter
+        cursorId={presenterCursorId._id}
+        {...rest}
+      />
+        : null }
+
+    {multiUserCursorIds && multiUserCursorIds.length > 0 ?
+          multiUserCursorIds.map(cursorId =>
+            (<CursorContainer
+              key={cursorId._id}
+              cursorId={cursorId._id}
+              presenter={false}
+              {...rest}
+            />),
+          )
+        : null }
+  </g>
+);
+
+export default createContainer(() => {
+  const { presenterCursorId, multiUserCursorIds } = CursorWrapperService.getCurrentCursorIds();
+  const isMultiUser = CursorWrapperService.getMultiUserStatus();
+
+  return {
+    presenterCursorId,
+    multiUserCursorIds,
+    isMultiUser,
+  };
+}, CursorWrapperContainer);
+
+
```

Comment:
I didn't understand this, you define the props as Object and Array, but their default value its undefined.

Because of this you need to keep checking for undefined AND for the value, you might change this to:

```
CursorWrapperContainer.defaultProps = {
   presenterCursorId: {},
   multiUserCursorIds: [],
 };
```

Or ask for them be required.

Reply:
I need to check if the presenter's cursor object is not empty anyway, otherwise it will try to render the cursor and throw exceptions in the process.
But that's definitely an improvement for the array, changed it 👍